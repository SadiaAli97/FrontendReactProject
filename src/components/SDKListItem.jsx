import * as React from 'react';
import { SPACE } from '../constants/app-constants';
import moment from 'moment';
import '../css/sdklistitem.css';

export function SDKListItem({sdkCategory, categoryItems = []}) {
  return (
    <div className='list-item'>
      <div className='category'>
        {sdkCategory}
      </div>
      <div className='items'>
        {
          categoryItems.map(item => {
            return <div className='item'>
              {item.name + SPACE + moment(item.lastSeenDate, "YYYYMMDD").fromNow()}
            </div>
          })
        }
      </div>
    </div>
  );
}