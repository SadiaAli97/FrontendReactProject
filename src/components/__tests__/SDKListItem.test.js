import TestRenderer from 'react-test-renderer';
import { SDKListItem } from '../SDKListItem';

it('renders SDKListItem ', () => {
	const testRenderer = TestRenderer.create(
		<SDKListItem
		  sdkCategory='category'
		  categoryItems={[{name:'name1'}]}
		/>
	);
  expect(testRenderer.toJSON()).toMatchSnapshot()
});

