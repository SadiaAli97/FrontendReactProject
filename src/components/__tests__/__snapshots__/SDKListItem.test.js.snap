// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SDKListItem  1`] = `
<div
  className="list-item"
>
  <div
    className="category"
  >
    category
  </div>
  <div
    className="items"
  >
    <div
      className="item"
    >
      name1 Invalid date
    </div>
  </div>
</div>
`;
